javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/muzzle/Reference.java

```diff
@@ -295,62 +296,63 @@ String getMismatchDetails() {
 
     public static class MissingMethod extends Mismatch {
       private final String className;
-      private final String method;
+      private final String methodSignature;
 
-      public MissingMethod(Source[] sources, String className, String method) {
+      public MissingMethod(Source[] sources, String className, String methodSignature) {
         super(sources);
         this.className = className;
-        this.method = method;
+        this.methodSignature = methodSignature;
       }
 
       @Override
       String getMismatchDetails() {
-        return "Missing method " + className + "#" + method;
+        return "Missing method " + className + "#" + methodSignature;
       }
     }
   }
 
   /** Expected flag (or lack of flag) on a class, method, or field reference. */
   public enum Flag {
+    // The following constants represent the exact visibility of a referenced class/method
```

Comment:
I thought about extracting a common `Flag` interface and making all these new enums implement it, but leaving them all in a single enum set. I like the separate fields idea because of how easy it'd be to detect merge conflicts (you can't have a field that is both static and non-static).

I'll do that in a separate PR - I'll refactor, clean up and document muzzle someday soon.